packages/compiler/server/serverlib.ts

```diff
@@ -613,25 +613,26 @@ export function createServer(host: ServerHost): Server {
     }
   }
 
-  async function getTypeDetails(type: Type): Promise<string> {
-    return type.kind;
+  async function getTypeDetails(sym: Sym, type: Type, program: Program): Promise<string> {
+    const doc = getDoc(program, type);
+    if (doc) {
+      return [type.kind, sym.name, doc].join(", ");
```

Comment:
Try to format this nicer than comma separated list. Hover around TypeScript files for inspiration.

Suggestions:
* Make it look like the Cadl declaration syntax, convert the kind to the appropriate lowercase keyword.
* Get the fully-qualified name of symbol.
* Put the documentation on a different line with some separation.

So, for example, if you had this in Cadl:

```
namespace Some.Namespace;
@doc("Some example")
model Example {}
```

You'd get something like this when you hovered over references to Example.

```
interface Some.Namespace.Example;

This is an example
```

Here is TypeScript.

![image](https://user-images.githubusercontent.com/75470/185426168-c9c2f861-8e20-4ee6-947e-19e9b006efaf.png)

Bonus points if you get the coloring and that horizontal separator too. :)

Reply:
For the coloring, I wonder if it works to put the declaration in a code region like this if you set the markup to markdown.

````
```cadl
interface Some.Namespace.Example;
```
---
This is an example
````